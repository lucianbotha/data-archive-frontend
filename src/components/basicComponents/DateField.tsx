import * as React from "react";

interface IDAInputProps {
  name?: string | undefined;
  value?: string | undefined;
  error?: string | undefined;
  onChange?: (e: React.FormEvent<HTMLInputElement>) => void;
  placeholder?: string | undefined;
}

const DateField = (props: IDAInputProps) => {
  const { value, onChange, error, name, placeholder } = props;
  return (
    <div className="control">
      <input
        title={error}
        id={`${name}`}
        type={"date"}
        className={`is-label input  ${error && error !== "" && "is-danger"}`}
        name={name}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
      />
      <p style={{ color: "red" }}>{error}</p>
    </div>
  );
};
export default DateField;
