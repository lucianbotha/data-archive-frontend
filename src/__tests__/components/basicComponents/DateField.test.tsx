import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import DateField from "../../../components/basicComponents/DateField";

describe("InpuField", () => {
  const wrapper = shallow(<DateField />);
  it("should render", () => {
    expect(wrapper).toBeDefined();
  });

  it("always renders a div", () => {
    const divs = wrapper.find("div");
    expect(divs.length).toBeGreaterThan(0);
  });
  it("always contains an input tag with of type date", () => {
    const inputs = wrapper.find("input");
    expect(inputs.length).toEqual(1);
    const input = wrapper.find("input").get(0);
    expect(input.props.type).toEqual("date");
  });
  it("render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
      toJson(shallow(<DateField error={"it has error"} />))
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(<DateField name="name" value="value" placeholder="holder" />)
      )
    ).toMatchSnapshot();
  });
});
