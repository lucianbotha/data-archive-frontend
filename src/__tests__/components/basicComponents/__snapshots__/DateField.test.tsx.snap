// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InpuField render correctly 1`] = `
<div
  className="control"
>
  <input
    className="is-label input  undefined"
    id="undefined"
    type="date"
  />
  <p
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;

exports[`InpuField render correctly 2`] = `
<div
  className="control"
>
  <input
    className="is-label input  is-danger"
    id="undefined"
    title="it has error"
    type="date"
  />
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    it has error
  </p>
</div>
`;

exports[`InpuField render correctly 3`] = `
<div
  className="control"
>
  <input
    className="is-label input  undefined"
    id="name"
    name="name"
    placeholder="holder"
    type="date"
    value="value"
  />
  <p
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;
